internal/pkg/cache/cache.go

```diff
@@ -270,3 +274,37 @@ func (c *CacheT) SetArtifact(artifact model.Artifact) {
 		Dur("ttl", ttl).
 		Msg("Artifact cache SET")
 }
+
+func (c *CacheT) SetUpload(id string, info upload.Info) {
+	c.mut.RLock()
+	defer c.mut.RUnlock()
+
+	scopedKey := "upload:" + id
+	ttl := time.Hour / 2 // @todo: add to configurable
+	cost := int64(len(info.ID) + len(info.DocID) + len(info.ActionID) + len(info.AgentID) + len(info.Source) + len(info.Status) + 8*4)
+	ok := c.cache.SetWithTTL(scopedKey, info, cost, ttl)
+	log.Trace().
+		Bool("ok", ok).
+		Str("id", id).
+		Int64("cost", cost).
+		Dur("ttl", ttl).
+		Msg("Upload info cache SET")
+}
+func (c *CacheT) GetUpload(id string) (upload.Info, bool) { //nolint:dupl // a little repetition to support strong typing
+	c.mut.RLock()
+	defer c.mut.RUnlock()
+
+	scopedKey := "upload:" + id
```

Comment:
nit: maybe use a function that makes the key string to avoid doing this in two places, easier to make mistake if anything changes.